webextensions/api/devtools.json

```
@@ -611,6 +914,65 @@
               }
             }
           },
+          "SourcesPanel": {
+            "__compat": {
+              "support": {
+                "chrome": {
+                  "version_added": true
+                },
+                "edge": "mirror",
+                "firefox": {
+                  "version_added": false
+                },
+                "firefox_android": "mirror",
+                "opera": "mirror",
+                "safari": {
+                  "version_added": false
+                },
+                "safari_ios": "mirror"
+              }
+            },
+            "createSidebarPane": {
+              "__compat": {
+                "mdn_url": "https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/devtools/panels/SourcesPanel/createSidebarPane",
+                "support": {
+                  "chrome": {
+                    "version_added": true
```

createSidebarPane is referenced in [this commit](https://source.chromium.org/chromium/chromium/src/+/e975d11d11145ede2784fc263355a3fde8eb8f71) for [chrome/common/extensions/api/devtools/panels.json](https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/extensions/api/devtools/panels.json;l=23;drc=5b4d80269c0b3032288cb54357bbf86ee2b787a7). While this commit comes from a time before what the [ChromiumDash commit tool](https://chromiumdash.appspot.com/commits) is able to look up, [this comment](https://issues.chromium.org/issues/40220015#comment79) on the associated bug indicates that was pushed out on the site on Aug 28, 2012. Using [this table](https://google.fandom.com/wiki/Chrome_version_history) of historical Chrome releases, it appears that the first release after that date was Chrome 22.0.1229. As such, that's my best guess for when this was first introduced.

```suggestion
                    "version_added": "22"
```

---

Good guess. Note that the source link you shared shows the (short) commit hash in the upper left corner of the UI. You can copy that and paste it in this URL:

`https://chromium.googlesource.com/chromium/src/+/<hash here>/chrome/VERSION`

https://chromium.googlesource.com/chromium/src/+/e975d11/chrome/VERSION


Note that your URL itself also contained the (long) hash: https://source.chromium.org/chromium/chromium/src/+/e975d11d11145ede2784fc263355a3fde8eb8f71

You could also have copied that to generate this identical URL:
https://chromium.googlesource.com/chromium/src/+/e975d11d11145ede2784fc263355a3fde8eb8f71/chrome/VERSION

---

I took a closer look at your commit. It explicitly states that the commit is about splitting an existing file, so 22 is likely not the right version.

---

To find the right version, I did the following:

- in the commit you linked, the parent commit is mentioned, `80512b8`.
- the commit message said that it split devtools_api.json
- I put used this commit hash and the file as yhe basis for the URL, and used `+blame` instead of `+` to show the blame history next to the code.

The resulting URL is: https://chromium.googlesource.com/chromium/src/+blame/60512b8/chrome/common/extensions/api/devtools_api.json

The blame points to https://chromium.googlesource.com/chromium/src/+/67b6d9c195af0461d173afa28b7188a5b1783bfc which was [version 16](https://chromium.googlesource.com/chromium/src/+/67b6d9c195af0461d173afa28b7188a5b1783bfc/chrome/VERSION).

BUT it was under the chrome.experimental.devtools.inspectedWindow namespace, so we went too far back.

If I scroll up again to look for the namespace definition, I can find that the namespace was changed in https://chromium.googlesource.com/chromium/src/+/2cf725f68fa5106fd5bcb1e1fb8699bb3628cafe , aka [version 18](https://chromium.googlesource.com/chromium/src/+/2cf725f68fa5106fd5bcb1e1fb8699bb3628cafe/chrome/VERSION).